src/python/yaramod_python.cpp

```diff
@@ -297,6 +297,9 @@ void addExpressionClasses(py::module& module)
 	py::class_<Expression, std::shared_ptr<Expression>>(module, "Expression")
 		.def("accept", &Expression::accept)
 		.def("get_text", &Expression::getText, py::arg("indent") = std::string{})
+		.def("exchange_tokens", [](Expression& self, Expression* other) {
```

Comment:
Won't `&Expression::exchangeTokens` be sufficient here instead of lambda?

Reply:
The `Expression::exchangeTokens` method has four overloads but I can put there this: 
`py::overload_cast<Expression*>(&Expression::exchangeTokens)`

Reply:
As for the second thing, I just pushed some [commits](https://github.com/avast/yaramod/pull/100/commits/79fde8d440be9192a6e0ca755a7dd6ff7bd60a46) with new `OrExpressionSwitcher` modifying visitor which may be a solution to this. This visitor has been written in a manner in which the dealing with tokens can be made 1. before and 2. at-the-end of the `visit` action. I will test this little further.